lib/index.js

```diff
@@ -66,30 +73,37 @@ var bootStrap = function (options) {
     gardr.id = gardr.params.id;
 
     gardr.log = logger.create(gardr.id, gardr.params.loglevel, getAppender(gardr.params.logto));
-
-    // TODO requestAnimationFrame polyfill
-
-    gardr.log.debug('Loading url: ' + gardr.params.url);
-
-    document.write(['<div id="gardr"><scr', 'ipt src="', gardr.params.url, '" ></scr', 'ipt></div>'].join(''));
-
-    gardr.container = document.getElementById('gardr');
-    gardr.container.style.overflow = 'hidden'; // avoid iOS Safari bug http://stackoverflow.com/q/6721310
-    pluginApi.trigger('element:containercreated', gardr.container);
-
-    var com = comClient(gardr.id, window.parent, gardr.params.origin);
-    eventListener.add(global, 'load', function () {
+    // IE 8 and 9 will not write into correct container (!) if we document.write directly from script-file-context.
+    gardr.inject = function inject() {
+        gardr.log.debug('Loading url: ' + gardr.params.url);
+        var str = ['<scr', 'ipt id="gardr-injected-script" type="text/javascript" src="',
+            gardr.params.url, '"></scr', 'ipt>'].join('');
+        doc.write(str);
+        gardr.container = doc.getElementById('gardr-injected-script').parentNode;
+        pluginApi.trigger('element:containercreated', gardr.container);
+    };
+
+    var com = comClient(gardr.id, win.parent, gardr.params.origin);
+    eventListener.add(global, 'load', function onloadDelayedHandler() {
         // phantomjs doesn't calculate sizes correctly unless we give it a break
-        setTimeout(function () {
+        timer(function () {
+            // parent.console.log(document.body.innerHTML);
             var size = childrenSize(gardr.container);
             pluginApi.trigger('banner:rendered', size);
             com.rendered(size);
         }, 0);
     });
+
+    return gardr;
 };
 
-bootStrap._setComClient = function (client) {
-    comClient = client;
+bootStrap._mock = function (_doc, _win, _timer) {
+    doc = _doc;
+    win = _win;
+    timer = _timer;
+    return function(){
+        timer = timeoutWrap;
```

Comment:
Readability and clean code is more important than number of files. We should try to avoid test-specific code outside the tests. It's also not very clear what the returned function's purpose is. I suppose it's for restoring the original function, but that code should be inside the tests.
By the way; Why not just replace window.setTimeout from the tests instead of wrapping?